### PR TITLE
WIP: MINIFICPP-1279 - FlowFile shouldn't inherit from Connectable

### PR DESCRIPTION
**Pull request by arpadboda**
Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [ ] Does your PR title start with MINIFICPP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by arpadboda** [comment]
This is tagged for 1.0.0 and WIP, please do NOT merge under any circumstance!

File: extensions/rocksdb-repos/FlowFileRepository.cpp [hunk 1]
```diff
@@ -152,7 +153,10 @@ void FlowFileRepository::prune_stored_flowfiles() {
         // we find the connection for the persistent flowfile, create the flowfile and enqueue that
         std::shared_ptr<core::FlowFile> flow_file_ref = std::static_pointer_cast<core::FlowFile>(eventRead);
         eventRead->setStoredToRepository(true);
-        search->second->put(eventRead);
+        auto connection = std::static_pointer_cast<Connection>(search->second);
```
Comment (adamdebreceni) [inline extensions/rocksdb-repos/FlowFileRepository.cpp:156]: if the #807 is merged Processors might also be FlowFile containers (like `BinFiles`) and those files should also be restored

**Comment by hunyadi-dev** [comment]
Minor, but this PR is only named WIP, but is not a draft. Also targets `master`.

**Comment by arpadboda** [comment]
> Minor, but this PR is only named WIP, but is not a draft. Also targets `master`.

Which makes me happy as it surely cannot be merged. :) It is just a POC, will be closed soon, should be properly done in #839

**Comment by arpadboda** [comment]
Closing, this was just a POC, will be properly done in #839

**Pull request closed without merging**